src/frontend/js/components/SaleTunnelStepPayment/index.tsx

```diff
@@ -0,0 +1,275 @@
+    return a.title.localeCompare(b.title, [intl.locale, intl.defaultLocale]);
+  };
+
+  const intl = useIntl();
+  const { user } = useSession();
+  const addresses = useAddresses();
+  const creditCards = useCreditCards();
+
+  const [showAddressCreationForm, setShowAddressCreationForm] = useState(false);
+
+  const [creditCard, setCreditCard] = useState<Maybe<Nullable<string>>>(
+    creditCards.items.find((c) => c.is_main)?.id,
+  );
+
+  const [address, setAddress] = useState<Maybe<Joanie.Address>>(() =>
+    addresses.items.find((a) => a.is_main),
+  );
+
+  const selectedCreditCard = useMemo(
+    () => (creditCard !== undefined ? creditCard : creditCards.items.find((c) => c.is_main)?.id),
+    [creditCard, creditCards],
+  );
+
+  const selectedAddress = useMemo(
+    (): Maybe<Joanie.Address> =>
+      address ||
+      addresses.items.find((a) => a.is_main) ||
+      [...addresses.items].sort(sortAddressByTitleAsc)[0],
+    [address, addresses],
+  );
+
+  /**
+   * Memoized address items.
+   * If the selected address is a local one, we have to add it to the item list.
+   * Finally, we sort address ascending by title before return them
+   *
+   * @returns {Joanie.Address[]} Sorted addresses ascending by title
+   */
+  const addressesItems = useMemo(() => {
+    const items = [...addresses.items];
+
+    // Add local address to the address item list.
+    if (selectedAddress?.id === LOCAL_BILLING_ADDRESS_ID) {
+      items.push(selectedAddress);
+    }
+
+    return items.sort(sortAddressByTitleAsc);
+  }, [addresses.items, selectedAddress]);
+
+  /**
+   * Retrieve address through its `id` provided by the event target value
+   * then update `selectedAddress` state
+   *
+   * @param {React.ChangeEvent<HTMLSelectElement>} event
+   */
+  const handleSelectAddress = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const targetedAddress = addresses.items.find((a) => a.id === event.target.value);
+    if (targetedAddress) {
+      setAddress(targetedAddress);
+    }
+  };
+
+  const toggleCreditCard = (creditCardId: string) => {
+    if (selectedCreditCard === creditCardId) {
+      setCreditCard(null);
+    } else {
+      setCreditCard(creditCardId);
+    }
+  };
+
+  if (showAddressCreationForm) {
+    return (
+      <AddressesManagement
+        handleClose={() => setShowAddressCreationForm(false)}
+        selectAddress={setAddress}
+      />
+    );
+  }
+
+  if (!user) return null;
+
+  return (
+    <section className="SaleTunnelStepPayment">
+      <section className="SaleTunnelStepPayment__block">
+        <header className="SaleTunnelStepPayment__block__header">
+          <h5 className="SaleTunnelStepPayment__block__title">
+            <FormattedMessage {...messages.resumeTile} />
+          </h5>
+        </header>
+        <div className="SaleTunnelStepPayment__block--product">
+          <h6 className="SaleTunnelStepPayment__block--product__title">{product.title}</h6>
```

Comment:
Seems weird to me that this is a heading. It doesn't really section content, it's content itself.